### PR TITLE
[f-nested-deps] Remove requirement to have root pack name in overrides

### PR DESCRIPTION
**Pull request by angrycub**
Continues on the work in #403. Removes the requirement to specify the name of the root pack in the override variable's path.

### CODE REVIEW

File: internal/pkg/varfile/varfile.go [hunk 1]
```diff
@@ -13,13 +13,15 @@ import (
 	"github.com/hashicorp/nomad-pack/sdk/pack/variables"
 )
 
-func DecodeVariableOverrides(files []*pack.File) DecodeResult {
+// TODO: this is only used in a test - remove?
```
Comment (pkazmierczak) [inline internal/pkg/varfile/varfile.go:16]: Remove.

File: internal/pkg/varfile/varfile.go [hunk 2]
```diff
@@ -217,9 +221,30 @@ func decode(filename string, src []byte, ctx *hcl.EvalContext, target *variables
 		oRange := hcl.RangeBetween(kv.Key.Range(), kv.Value.Range())
 		fixupRange(&oRange)
 
+		var path pack.ID
+		var name variables.ID
+
+		// NOTE: This implementation assumes a single element variable value,
```
Comment (pkazmierczak) [inline internal/pkg/varfile/varfile.go:227]: I think that's fine for now but maybe make an issue so that we could revisit later?

**Review by pkazmierczak — APPROVED** [review]
LGTM!

File: internal/pkg/varfile/varfile.go (hunk 2 above)
Comment (angrycub) [reply internal/pkg/varfile/varfile.go:227]: Created issue #435 to capture this

File: internal/pkg/varfile/varfile.go (hunk 1 above)
Comment (angrycub) [reply internal/pkg/varfile/varfile.go:16]: Moved into the test package

**Pull request merged**